var/spack/repos/builtin/packages/namd/package.py

```diff
@@ -104,6 +104,8 @@ class Namd(MakefilePackage, CudaPackage):
     depends_on("python", when="interface=python")
 
     conflicts("+avxtiles", when="@:2.14,3:", msg="AVXTiles algorithm requires NAMD 2.15")
+    conflicts("+rocm", when="~single_node_gpu")
```

Comment:
I don't understand where this conflict is coming from. The NAMD config script definitely suggests that --with-hip can be used without --with-single-node-hip.

Reply:
Is there a way to specify that +single_node_gpu requires either +cuda or +rocm? I can't find the "when" syntax in the docs.

Reply:
I suggested the conflict because the original implementation only allowed for `+single_node_gpu`. But with the new implementation, the conflict is not necessary since both variants are supported.

Reply:
What about this?

```
conflicts("+single_node_gpu", when="~cuda~rocm")
```